### ISSUE TITLE
Minimize the number of times the `cubic.conf` project configuration file is saved.

### ISSUE THREAD

**Issue by PJ-Singh-001** [PJ-Singh-001/Cubic#171]
Currently the `cubic.conf` project configuration file is saved at least five times while using Cubic. The file is saved when transitioning to the next page from each of the following pages: Compression page, Extract page, Generate page, Options page, and Project page.

If the user navigates back and then forward, the configuration file will be saved again, for each of the above transitions. On the Options page, the file is also saved when navigating back to the Terminal page. The file is saved whether or not changes have been made. Although the file is small, this still represents unnecessary disk writes.

**Comment by PJ-Singh-001** [comment]
To minimize the number of times the `cubic.conf` project configuration file is written to disk, only save the file when quitting Cubic. Changes to the configuration should be stored in memory while using Cubic.

Thus, on each page, save the `cubic.conf` project configuration on the quit action. 

As an exception, the configuration file will not be saved on the quit action for the Start page or the Project page, because the user must acknowledge changes on these pages by navigating from the Project page to the Extract page or the Terminal page. As a consequence, if the user returns to the Start page without quitting, all configuration changes will be overwritten by defaults/blanks. To address this, the configuration file must be saved whenever the user navigates back to the Project page from the Extract page or the Terminal page.

In summary, the configuration file will be saved only once for each run, with the exception that it will also be saved any time time the user returns back to the Project page.

(Note, the application configuration file, `cubic.conf` located in the user's home directory, is already only saved once when the application quits).

**Comment by PJ-Singh-001** [comment]
Places where saving the project configuration file has been removed, and places where saving the project configuration file has been added.

    Boot Copy Page
    - Added on leave/quit

    Compression Page
    - Removed from leave/back, leave/generate, leave/quit, leave/unknown
    - Added on leave/quit

    Delete Page
    - N/A

    Extract Page
    - Removed from leave/back, leave/next, leave/quit, leave/unknown

    Finish Page
    - Added on leave/close
    - Added on leave/quit

    Generate Page
    - Removed from leave/back, leave/finish, leave/quit, leave/unknown
    - Added on leave/quit

    Options Page
    - Removed from leave/back, leave/copy-preseed, leave/copy-boot, leave/back-terminal, leave/next, leave/quit, leave/unknown
    - Added on leave/quit

    Packages Page
    - Added on leave/quit

    Prepare Page
    - Added on leave/quit

    Preseed Copy Page
    - Added on leave/quit

    Project Page
    - Removed from leave/next, leave/next-terminal
    - Not on leave/quit, so no need to remove
    - Added on setup/back

    Start Page
    - N/A

    Terminal Copy Page
    - Added on leave/quit

    Terminal Page
    - Added on leave/quit

    Test 1 Page
    - Added on leave/quit

    Test 2 Page
    - Added on leave/quit

**Comment by PJ-Singh-001** [comment]
Fix released in version 2023.03.76.
